core/src/scene/styleContext.cpp

```diff
@@ -432,10 +432,7 @@ void StyleContext::parseStyleResult(StyleParamKey _key, StyleParam::Value& _val)
                     duk_pop(m_ctx);
                 }
 
-                _val = (((uint32_t)(255.0 * a) & 0xff) << 24) |
-                       (((uint32_t)(255.0 * r) & 0xff)<< 16) |
-                       (((uint32_t)(255.0 * g) & 0xff)<< 8) |
-                       (((uint32_t)(255.0 * b) & 0xff));
+                _val = Color::fromNormalized(r, g, b, a).abgr;
```

Comment:
I could be completely missing something here!

You are initializing the `Color`'s anonymous union with `r`, `g`, `b`, `a` constructor then I am not sure how you can access `abgr` from the anonymous union?

Reply:
Setting the `uint8_t` members of the union to the component values initializes the memory occupied by the `uint32_t` member in the same way as packing them with bit shifting, like this code was trying to do. Strictly speaking, this only works on little-endian systems, but we don't currently target (or project to target) big-endian systems.

Reply:
Ahh make sense, I would however refrain more such memory trickery and provide a method to do the apt conversion.

To quote the documentation "The details of that allocation are implementation-defined, and it's undefined behavior to read from the member of the union that wasn't most recently written". http://en.cppreference.com/w/cpp/language/union

Reply:
It actually gets interesting, https://stackoverflow.com/questions/11373203/accessing-inactive-union-member-and-undefined-behavior

Reply:
I got this trick from code for a game and it's always seemed to work, but you may be right! It's not totally obvious whether this is UB: https://stackoverflow.com/questions/11373203/accessing-inactive-union-member-and-undefined-behavior
In short: this trick is allowed in C, and the C++ spec doesn't outright forbid it but there are only a few cases where it isn't UB.